ethsigner/signer/hashicorp/src/main/java/tech/pegasys/ethsigner/signer/hashicorp/HashicorpSignerFactory.java

```diff
@@ -45,6 +45,21 @@
   private static final String TIMEOUT_MESSGAE =
       "Timeout while retrieving private key from Hashicorp Vault.";
 
+  public TransactionSigner createSigner(HashicorpConfig hashicorpConfig) {
+    final String response =
+        requestSecretFromVault(
+            hashicorpConfig.getSigningKeyPath(),
+            hashicorpConfig.getPort(),
+            hashicorpConfig.getHost(),
+            hashicorpConfig.getAuthFilePath(),
+            hashicorpConfig.getTimeout());
+    final Credentials credentials = extractCredentialsFromJson(response);
+    if (credentials.getAddress() != null) {
+      LOG.debug("Successfully retrieved the credentials from the Hashicorp vault.");
+    }
+    return new CredentialTransactionSigner(credentials);
+  }
+
   public static TransactionSigner createSigner(
```

Comment:
can we remove this, and only have the config based function - will need to update the HashicorpSubCOmmand to create a config object ... but it looks like you've already done that

Reply:
Had wondered about that.  Will implement.